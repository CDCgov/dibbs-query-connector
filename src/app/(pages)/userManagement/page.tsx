"use client";
import classNames from "classnames";
import { UserRole } from "@/app/models/entities/users";
import WithAuth from "@/app/ui/components/withAuth/WithAuth";
import UserManagementProvider from "./components/UserManagementProvider";
import UserManagementContainer from "./components/userManagementContainer/userManagementContainer";
import { getRole, RoleDescriptons } from "./utils";

/**
 * @param textContent the text values to display
 * @returns HTML element containing text descriptions for each of the User roles
 */
const renderRoleDescriptions = (textContent: object) => {
  return (
    <div
      className={classNames(
        "margin-x-3",
        "margin-y-4",
        "grid-row",
        "flex-row",
        "flex-no-wrap",
        "flex-justify",
        "grid-gap",
      )}
      aria-description="Available user roles"
    >
      {Object.entries(textContent).map(([role, description]) => (
        <p key={`${role}`} className={classNames("grid-col-4")}>
          <span className="text-bold">{`${role}:`}</span> {description}
        </p>
      ))}
    </div>
  );
};

/**
 * Client side parent component for the User Management page
 * @returns the UserManagement component
 */
const UserManagement: React.FC = () => {
  const role = getRole();

  return (
    <WithAuth>
      <div className={classNames("main-container__wide", "user-management")}>
        <div
          className={classNames(
            "grid-row",
            "padding-0",
            "flex-justify",
            "flex-align-center",
          )}
        >
          <h1 className="page-title">User management</h1>
        </div>
        <UserManagementProvider>
          {role == UserRole.SUPER_ADMIN &&
            renderRoleDescriptions(RoleDescriptons)}
          <UserManagementContainer role={role as string} />
        </UserManagementProvider>
      </div>
    </WithAuth>
  );
};

export default UserManagement;
