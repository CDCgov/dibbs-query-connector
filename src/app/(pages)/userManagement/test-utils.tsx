import React, { createRef, RefObject } from "react";
import { Tab } from "@/app/ui/designSystem/TabGroup/tabGroup";
import { UserRole, User, UserGroup } from "@/app/models/entities/users";
import UserPermissionsTable from "./components/userPermissionsTable/userPermissionsTable";
import UserGroupsTable from "./components/userGroupsTable/UserGroupsTable";
import { CustomUserQuery } from "@/app/models/entities/query";

// User Mocks
export const mockSuperAdmin = {
  id: "123",
  username: "harrypotter",
  firstName: "Harry",
  lastName: "Potter",
  qcRole: UserRole.SUPER_ADMIN,
};

export const mockAdmin: User = {
  id: "456",
  username: "lilypotter",
  firstName: "Lily",
  lastName: "Potter",
  qcRole: UserRole.ADMIN,
  userGroupMemberships: [
    {
      membershipId: "456_876",
      usergroupName: "Order of the Phoenix",
      usergroupId: "876",
      isMember: true,
    },
  ],
};

export const mockStandard: User = {
  id: "789",
  username: "hermionegranger",
  firstName: "Hermione",
  lastName: "Granger",
  qcRole: UserRole.ADMIN,
  userGroupMemberships: [
    {
      membershipId: "789_012",
      usergroupName: "S.P.E.W.",
      usergroupId: "012",
      isMember: true,
    },
  ],
};

// UserGroup Mocks
export const mockGroupBasic: UserGroup = {
  id: "123",
  name: "Hogwarts",
  memberSize: 1,
  querySize: 0,
};

export const mockGroupMany: UserGroup = {
  id: "456",
  name: "Order of the Phoenix",
  memberSize: 3,
  querySize: 3,
  members: [{} as User, {} as User, {} as User],
  queries: [
    {} as CustomUserQuery,
    {} as CustomUserQuery,
    {} as CustomUserQuery,
  ],
};

const mockGroupSingle = {
  id: "789",
  name: "S.P.E.W.",
  memberSize: 1,
  querySize: 1,
  members: [mockAdmin],
  queries: [{} as CustomUserQuery],
};

export const allGroups = [mockGroupMany, mockGroupSingle];

const mockRefs = createRef<RefObject<HTMLTableRowElement | null>[]>();
mockRefs.current = [];

// TabGroup Mocks
export const mockPermissionsTab: Tab = {
  label: "Users",
  access: ["Super Admin"],
  onClick: jest.fn(),
  renderContent: jest
    .fn()
    .mockReturnValue(
      <UserPermissionsTable
        setUsers={jest.fn()}
        users={[mockAdmin, mockSuperAdmin]}
        fetchGroupMembers={jest.fn()}
        openModal={jest.fn()}
        rowFocusRefs={mockRefs}
        modalData={""}
      />,
    ),
};
export const mockGroupsTab: Tab = {
  label: "User Groups",
  access: ["Super Admin", "Admin"],
  onClick: jest.fn(),
  renderContent: jest
    .fn()
    .mockReturnValue(
      <UserGroupsTable
        userGroups={[mockGroupBasic]}
        fetchGroupMembers={jest.fn().mockReturnValue(mockAdmin)}
        fetchGroupQueries={jest.fn()}
        openModal={jest.fn()}
        modalData={""}
      />,
    ),
};

// Query Mocks
export const mockQueryNoGroups: CustomUserQuery = {
  queryId: "q1",
  queryName: "Test Query 1",
  conditionsList: [],
  valuesets: [],
};

export const mockQueryWithGroups: CustomUserQuery = {
  queryId: "q2",
  queryName: "Test Query 2",
  conditionsList: [],
  valuesets: [],
  groupAssignments: [
    {
      membershipId: `q2_${mockGroupSingle.id}`,
      usergroupName: mockGroupSingle.name,
      usergroupId: mockGroupSingle.id,
      isMember: true,
    },
  ],
};

mockGroupSingle.queries.push(mockQueryWithGroups);

export const mockGroupWithSingleQuery = mockGroupSingle;
