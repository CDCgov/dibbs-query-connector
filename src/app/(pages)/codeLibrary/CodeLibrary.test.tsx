import { screen, render, act, waitFor } from "@testing-library/react";
import { RootProviderMock } from "@/app/tests/unit/setup";
import {
  gonorrheaValueSets,
  cancerValueSets,
  customValueSets,
  conditionIdToNameMap,
  categoryToConditionNameArrayMap,
} from "../queryBuilding/fixtures";
import CodeLibrary from "./page";
import {
  getAllValueSets,
  getConditionsData,
} from "@/app/shared/database-service";
import { getAllUsers, getUserByUsername } from "@/app/backend/user-management";
import {
  getAllGroupMembers,
  getAllUserGroups,
} from "@/app/backend/usergroup-management";
import { mockAdmin } from "../userManagement/test-utils";
import { getCustomValueSetById } from "@/app/shared/custom-code-service";

jest.mock("next-auth/react");

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: jest.fn(),
    prefetch: jest.fn(),
    replace: jest.fn(),
    back: jest.fn(),
    forward: jest.fn(),
    refresh: jest.fn(),
  }),
}));

jest.mock(
  "@/app/ui/components/withAuth/WithAuth",
  () =>
    ({ children }: React.PropsWithChildren) => <div>{children}</div>,
);

jest.mock("@/app/shared/database-service", () => ({
  getAllValueSets: jest.fn().mockReturnValue({ items: [] }),
  getConditionsData: jest.fn().mockReturnValue({}),
  groupConditionConceptsIntoValueSets: jest.fn().mockReturnValue([]),
}));

jest.mock("@/app/backend/usergroup-management", () => ({
  getAllUserGroups: jest.fn(),
  getAllGroupMembers: jest.fn(),
}));

jest.mock("@/app/backend/user-management", () => ({
  getAllUsers: jest.fn(),
  getUserByUsername: jest.fn(),
}));

jest.mock("@/app/shared/custom-code-service", () => ({
  getCustomValueSetById: jest.fn(),
}));

describe("Code library loading view", () => {
  beforeAll(() => {
    (getUserByUsername as jest.Mock).mockResolvedValue({ items: [mockAdmin] });
  });
  it("renders a skeleton loading state when loading is true", async () => {
    await act(async () =>
      render(
        <RootProviderMock currentPage="/codeLibrary">
          <CodeLibrary />
        </RootProviderMock>,
      ),
    );
    expect(await screen.findByTestId("loading-skeleton")).toBeInTheDocument();

    expect(document.body).toMatchSnapshot();
  });
});

describe("Code library rendered view", () => {
  const valueSets = [
    cancerValueSets,
    gonorrheaValueSets,
    customValueSets,
  ].flat();

  beforeAll(() => {
    (getAllUsers as jest.Mock).mockResolvedValue({ items: [], totalItems: 0 });
    (getAllGroupMembers as jest.Mock).mockResolvedValue({
      items: [],
      totalItems: 0,
    });
    (getUserByUsername as jest.Mock).mockResolvedValue({ items: [mockAdmin] });

    (getConditionsData as jest.Mock).mockResolvedValue({
      conditionIdToNameMap,
      categoryToConditionNameArrayMap,
    });

    (getAllValueSets as jest.Mock).mockReturnValue({ items: valueSets });

    (getCustomValueSetById as jest.Mock).mockReturnValue({
      items: customValueSets[0],
    });
  });

  it("renders the correct content", async () => {
    await act(async () =>
      render(
        <RootProviderMock currentPage="/codeLibrary">
          <CodeLibrary />
        </RootProviderMock>,
      ),
    );
    expect(screen.getByTestId("table-valuesets-manage")).toBeInTheDocument();

    const header = screen
      .getAllByRole("heading")
      .find((heading) => heading.classList.contains("header__title"));
    const infoAlert = screen.queryAllByTestId("alert")[0];

    expect(header).toHaveTextContent("Manage codes");
    expect(infoAlert).toHaveTextContent(
      "Value sets are an organizational structure for easy management of codes. Every code belongs to a value set.",
    );

    const tableBody = screen.getByTestId("table-valuesets-manage");
    expect(tableBody.childNodes[0]).toHaveTextContent(
      cancerValueSets[1].valueset_name,
    );
    expect(tableBody.childNodes[1]).toHaveTextContent(
      gonorrheaValueSets[1].valueset_name,
    );
    expect(tableBody.childNodes[2]).toHaveTextContent(
      customValueSets[1].valueset_name,
    );
    expect(document.body).toMatchSnapshot();
  });

  it("renders concept codes for the value set when clicked", async () => {
    await act(async () =>
      render(
        <RootProviderMock currentPage="/codeLibrary">
          <CodeLibrary />
        </RootProviderMock>,
      ),
    );

    expect(screen.getByTestId("table-valuesets-manage")).toBeInTheDocument();
    const tableBody = screen.getByTestId("table-valuesets-manage");
    () => (tableBody.firstChild as HTMLElement).click();

    expect(
      screen.getByText(
        "This value set comes from DIBBs and cannot be modified.",
      ),
    ).toBeVisible();

    expect(tableBody.childNodes).toHaveLength(3);
    expect(document.body).toMatchSnapshot();
  });
});

describe("Code library interaction", () => {
  const valueSets = [
    cancerValueSets,
    gonorrheaValueSets,
    customValueSets,
  ].flat();
  beforeAll(() => {
    (getAllUsers as jest.Mock).mockResolvedValue({ items: [], totalItems: 0 });
    (getAllUserGroups as jest.Mock).mockResolvedValue({
      items: [],
      totalItems: 1,
    });

    (getAllGroupMembers as jest.Mock).mockResolvedValue({
      items: [],
      totalItems: 0,
    });
    (getUserByUsername as jest.Mock).mockResolvedValue({ items: [mockAdmin] });

    (getConditionsData as jest.Mock).mockResolvedValue({
      conditionIdToNameMap,
      categoryToConditionNameArrayMap,
    });

    (getAllValueSets as jest.Mock).mockReturnValue({ items: valueSets });

    (getCustomValueSetById as jest.Mock).mockReturnValue({
      items: customValueSets[0],
    });
  });
  it("renders action buttons for custom value sets only", async () => {
    await act(async () =>
      render(
        <RootProviderMock currentPage="/codeLibrary">
          <CodeLibrary />
        </RootProviderMock>,
      ),
    );
    expect(screen.getByTestId("table-valuesets-manage")).toBeInTheDocument();

    const tableBody = screen.getByTestId("table-valuesets-manage");
    const cancerVsRow = tableBody.getElementsByClassName(
      "valueSetTable__tableBody_row_details",
    )[0];
    const customVsRow = tableBody.getElementsByClassName(
      "valueSetTable__tableBody_row_details",
    )[2];

    expect(cancerVsRow.textContent).toContain(cancerValueSets[0].valueset_name);
    expect(customVsRow.textContent).toContain(customValueSets[0].valueset_name);

    expect(cancerVsRow.textContent).not.toContain(cancerValueSets[0].author);
    expect(customVsRow.textContent).toContain(customValueSets[0].author);

    await waitFor(async () => {
      (cancerVsRow as HTMLElement).click();
      const cancerCodesPanel = screen.getByTestId("table-codes").parentElement;
      expect(cancerCodesPanel).toHaveTextContent(
        "This value set comes from DIBBs and cannot be modified.",
      );
    });

    await waitFor(async () => {
      (customVsRow as HTMLElement).click();
      const customCodesPanel = screen.getByTestId("table-codes").parentElement;

      const customActionButtons = (
        customCodesPanel as Element
      ).getElementsByClassName("usa-button");
      expect(customActionButtons[0]).toHaveTextContent("Edit codes");
      expect(customActionButtons[1]).toHaveTextContent("Delete value set");
    });

    expect(document.body).toMatchSnapshot();
  });

  it("prompts the user to confirm before deleting a value set", async () => {
    await act(async () =>
      render(
        <RootProviderMock currentPage="/codeLibrary">
          <CodeLibrary />
        </RootProviderMock>,
      ),
    );

    const customVsRow = screen
      .getByTestId("table-valuesets-manage")
      .getElementsByClassName("valueSetTable__tableBody_row_details")[2];

    await waitFor(async () => {
      (customVsRow as HTMLElement).click();
      const customCodesPanel = screen.getByTestId("table-codes").parentElement;

      const deleteButton = (customCodesPanel as Element).getElementsByClassName(
        "usa-button",
      )[1];

      (deleteButton as HTMLElement).click();

      const modal = screen.getByTestId("modalWindow");
      expect(modal).toBeVisible();
      expect(modal).toHaveTextContent(
        `Are you sure you want to delete the value set "${customValueSets[0].valueset_name}?"`,
      );
    });
  });
});
