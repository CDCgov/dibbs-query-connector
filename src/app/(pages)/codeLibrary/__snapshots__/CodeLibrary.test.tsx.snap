// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code library loading view renders a skeleton loading state when loading is true 1`] = `
<body>
  <div>
    <div>
      <div
        class="main-container__wide mainContainer"
      >
        <div
          class="header"
        >
          <a
            aria-label="Back arrow indicating ability to navigate back a page if clicked"
            class="back-link unchanged-color-on-visit text-no-underline"
            data-testid="backArrowLink"
            href="#"
          >
            <svg
              aria-label="Arrow point left indicating return to previous step"
              class="usa-icon"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
             
            Back to Query library
          </a>
          <h1
            class="header__title"
          >
            Manage codes
          </h1>
          <div
            class="header__subtitle"
          >
            Click on the checkbox to delete the value set or code
          </div>
          <div
            class="usa-alert usa-alert--info info-alert"
            data-testid="alert"
          >
            <div
              class="usa-alert__body"
            >
              <p
                class="usa-alert__text"
              >
                Value sets are an organizational structure for easy management of codes. Every code belongs to a value set.
              </p>
            </div>
          </div>
        </div>
        <div
          class="controls"
        >
          <div
            class="searchFilter"
          >
            <input
              class="usa-input searchField search"
              data-testid="textInput"
              id="librarySearch"
              name="librarySearch"
              placeholder="Search"
              type="search"
              value=""
            />
            <div
              class="applyFilters"
              role="button"
              tabindex="0"
            >
              <svg
                aria-label="Icon indicating a menu with filter options"
                class="usa-icon usa-icon--size-3 usa-icon qc-filter"
                focusable="false"
                height="1em"
                role="icon"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                />
              </svg>
              Filters
               
            </div>
          </div>
          <button
            class="usa-button button"
            data-testid="button"
            type="button"
          >
            Add value set
          </button>
        </div>
        <div
          class="content"
        >
          <div
            class="resultsContainer"
          >
            <div
              class="content__left"
            >
              <div
                class="display-flex flex-row valueSetTable customizeTableWrapper"
              >
                <table
                  class="usa-table usa-table--borderless customizeTable"
                  data-testid="table"
                >
                  <thead
                    class="display-flex flex-column valueSetTable__header"
                  >
                    <tr
                      class="valueSetTable__header_sectionHeader"
                    >
                      <th>
                        VALUE SET
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="overflowScroll valueSetTable__tableBody"
                  >
                    <tr
                      class="valueSetTable__tableBody_row"
                      data-testid="loading-skeleton"
                    >
                      <td>
                        <span
                          aria-busy="true"
                          aria-live="polite"
                          class="skeletonContainer"
                        >
                          <span
                            class="react-loading-skeleton skeleton"
                          >
                            ‌
                          </span>
                          <br />
                          <span
                            class="react-loading-skeleton skeleton"
                          >
                            ‌
                          </span>
                          <br />
                          <span
                            class="react-loading-skeleton skeleton"
                          >
                            ‌
                          </span>
                          <br />
                          <span
                            class="react-loading-skeleton skeleton"
                          >
                            ‌
                          </span>
                          <br />
                          <span
                            class="react-loading-skeleton skeleton"
                          >
                            ‌
                          </span>
                          <br />
                          <span
                            class="react-loading-skeleton skeleton"
                          >
                            ‌
                          </span>
                          <br />
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              class="content__right"
            />
          </div>
          <div
            class="paginationContainer"
          >
            <span>
              Showing 0 -
  0 of 
  0 value sets
            </span>
            <nav
              aria-label="Pagination"
              class="usa-pagination pagination"
            >
              <ul
                class="usa-pagination__list"
              >
                <li
                  class="usa-pagination__item usa-pagination__page-no"
                >
                  <button
                    aria-current="page"
                    aria-label="Page 1"
                    class="usa-button usa-button--unstyled usa-pagination__button usa-current"
                    data-testid="pagination-page-number"
                    type="button"
                  >
                    1
                  </button>
                </li>
              </ul>
            </nav>
            <div
              class="itemsPerPageContainer"
            >
              <label
                for="actionsPerPage"
              >
                Value sets per page
              </label>
              <select
                class="usa-select itemsPerPageDropdown"
                data-testid="Select"
                id="valeSetsPerPage"
                name="valeSetsPerPage"
              >
                <option
                  value="10"
                >
                  10
                </option>
                <option
                  value="25"
                >
                  25
                </option>
                <option
                  value="50"
                >
                  50
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Code library rendered view renders concept codes for the value set when clicked 1`] = `
<body>
  <div>
    <div>
      <div
        class="main-container__wide mainContainer"
      >
        <div
          class="header"
        >
          <a
            aria-label="Back arrow indicating ability to navigate back a page if clicked"
            class="back-link unchanged-color-on-visit text-no-underline"
            data-testid="backArrowLink"
            href="#"
          >
            <svg
              aria-label="Arrow point left indicating return to previous step"
              class="usa-icon"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
             
            Back to Query library
          </a>
          <h1
            class="header__title"
          >
            Manage codes
          </h1>
          <div
            class="header__subtitle"
          >
            Click on the checkbox to delete the value set or code
          </div>
          <div
            class="usa-alert usa-alert--info info-alert"
            data-testid="alert"
          >
            <div
              class="usa-alert__body"
            >
              <p
                class="usa-alert__text"
              >
                Value sets are an organizational structure for easy management of codes. Every code belongs to a value set.
              </p>
            </div>
          </div>
        </div>
        <div
          class="controls"
        >
          <div
            class="searchFilter"
          >
            <input
              class="usa-input searchField search"
              data-testid="textInput"
              id="librarySearch"
              name="librarySearch"
              placeholder="Search"
              type="search"
              value=""
            />
            <div
              class="applyFilters"
              role="button"
              tabindex="0"
            >
              <svg
                aria-label="Icon indicating a menu with filter options"
                class="usa-icon usa-icon--size-3 usa-icon qc-filter"
                focusable="false"
                height="1em"
                role="icon"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                />
              </svg>
              Filters
               
            </div>
          </div>
          <button
            class="usa-button button"
            data-testid="button"
            type="button"
          >
            Add value set
          </button>
        </div>
        <div
          class="content"
        >
          <div
            class="resultsContainer"
          >
            <div
              class="content__left"
            >
              <div
                class="display-flex flex-row valueSetTable customizeTableWrapper"
              >
                <table
                  class="usa-table usa-table--borderless customizeTable"
                  data-testid="table"
                >
                  <thead
                    class="display-flex flex-column valueSetTable__header"
                  >
                    <tr
                      class="valueSetTable__header_sectionHeader"
                    >
                      <th>
                        VALUE SET
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="overflowScroll valueSetTable__tableBody"
                  >
                    <tr
                      class="valueSetTable__tableBody_row activeValueSet"
                    >
                      <td>
                        <div
                          class="valueSetTable__tableBody_row_details"
                        >
                          <span
                            class="valueSetTable__tableBody_row_valueSetName"
                          >
                            <span
                              class=""
                            >
                              Cancer (Leukemia) Lab Result
                            </span>
                          </span>
                          <span
                            class="valueSetTable__tableBody_row_valueSetDetails"
                          >
                            <span
                              class=""
                            >
                              Cancer (Leukemia)  •   Labs  •  http://cap.org/eCC
                            </span>
                          </span>
                        </div>
                        <div>
                          <svg
                            aria-label="Right chevron indicating additional content"
                            class="usa-icon usa-icon--size-4"
                            focusable="false"
                            height="1em"
                            role="img"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                            />
                          </svg>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="valueSetTable__tableBody_row"
                    >
                      <td>
                        <div
                          class="valueSetTable__tableBody_row_details"
                        >
                          <span
                            class="valueSetTable__tableBody_row_valueSetName"
                          >
                            <span
                              class=""
                            >
                              Gonorrhea Medication
                            </span>
                          </span>
                          <span
                            class="valueSetTable__tableBody_row_valueSetDetails"
                          >
                            <span
                              class=""
                            >
                              Gonorrhea  •   Medications  •  RXNORM
                            </span>
                          </span>
                        </div>
                        <div>
                          <svg
                            aria-label="Right chevron indicating additional content"
                            class="usa-icon usa-icon--size-4"
                            focusable="false"
                            height="1em"
                            role="img"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                            />
                          </svg>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              class="content__right"
            >
              <div
                class="display-flex flex-row conceptsTable customizeTableWrapper"
              >
                <table
                  class="usa-table usa-table--borderless customizeTable"
                  data-testid="table"
                >
                  <thead
                    class="display-flex flex-column conceptsTable__header"
                  >
                    <tr
                      class="lockedForEdits"
                    >
                      <th>
                        <svg
                          class="usa-icon qc-lock"
                          focusable="false"
                          height="1em"
                          role="icon"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                          />
                        </svg>
                        This value set comes from the CSTE and cannot be modified.
                      </th>
                    </tr>
                    <tr
                      class="columnHeaders"
                    >
                      <th>
                        Code
                      </th>
                      <th>
                        Name
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="overflowScroll_headerLocked conceptsTable__tableBody"
                  >
                    <tr
                      class="conceptsTable__tableBody_row"
                    >
                      <td
                        class="valueSetCode"
                      >
                        <span>
                          <span
                            class=""
                          >
                            9484.100004300
                          </span>
                        </span>
                      </td>
                      <td>
                        <span>
                          <span
                            class=""
                          >
                            RESULTS
                          </span>
                        </span>
                      </td>
                    </tr>
                    <tr
                      class="conceptsTable__tableBody_row"
                    >
                      <td
                        class="valueSetCode"
                      >
                        <span>
                          <span
                            class=""
                          >
                            1255068005
                          </span>
                        </span>
                      </td>
                      <td>
                        <span>
                          <span
                            class=""
                          >
                            Presence of DNA mismatch repair protein MSH2 in primary malignant neoplasm of colon by immunohistochemistry (observable entity)
                          </span>
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
          <div
            class="paginationContainer"
          >
            <span>
              Showing 1 -
  2 of 
  2 value sets
            </span>
            <nav
              aria-label="Pagination"
              class="usa-pagination pagination"
            >
              <ul
                class="usa-pagination__list"
              >
                <li
                  class="usa-pagination__item usa-pagination__page-no"
                >
                  <button
                    aria-current="page"
                    aria-label="Page 1"
                    class="usa-button usa-button--unstyled usa-pagination__button usa-current"
                    data-testid="pagination-page-number"
                    type="button"
                  >
                    1
                  </button>
                </li>
              </ul>
            </nav>
            <div
              class="itemsPerPageContainer"
            >
              <label
                for="actionsPerPage"
              >
                Value sets per page
              </label>
              <select
                class="usa-select itemsPerPageDropdown"
                data-testid="Select"
                id="valeSetsPerPage"
                name="valeSetsPerPage"
              >
                <option
                  value="10"
                >
                  10
                </option>
                <option
                  value="25"
                >
                  25
                </option>
                <option
                  value="50"
                >
                  50
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Code library rendered view renders the correct content 1`] = `
<body>
  <div>
    <div>
      <div
        class="main-container__wide mainContainer"
      >
        <div
          class="header"
        >
          <a
            aria-label="Back arrow indicating ability to navigate back a page if clicked"
            class="back-link unchanged-color-on-visit text-no-underline"
            data-testid="backArrowLink"
            href="#"
          >
            <svg
              aria-label="Arrow point left indicating return to previous step"
              class="usa-icon"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
             
            Back to Query library
          </a>
          <h1
            class="header__title"
          >
            Manage codes
          </h1>
          <div
            class="header__subtitle"
          >
            Click on the checkbox to delete the value set or code
          </div>
          <div
            class="usa-alert usa-alert--info info-alert"
            data-testid="alert"
          >
            <div
              class="usa-alert__body"
            >
              <p
                class="usa-alert__text"
              >
                Value sets are an organizational structure for easy management of codes. Every code belongs to a value set.
              </p>
            </div>
          </div>
        </div>
        <div
          class="controls"
        >
          <div
            class="searchFilter"
          >
            <input
              class="usa-input searchField search"
              data-testid="textInput"
              id="librarySearch"
              name="librarySearch"
              placeholder="Search"
              type="search"
              value=""
            />
            <div
              class="applyFilters"
              role="button"
              tabindex="0"
            >
              <svg
                aria-label="Icon indicating a menu with filter options"
                class="usa-icon usa-icon--size-3 usa-icon qc-filter"
                focusable="false"
                height="1em"
                role="icon"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                />
              </svg>
              Filters
               
            </div>
          </div>
          <button
            class="usa-button button"
            data-testid="button"
            type="button"
          >
            Add value set
          </button>
        </div>
        <div
          class="content"
        >
          <div
            class="resultsContainer"
          >
            <div
              class="content__left"
            >
              <div
                class="display-flex flex-row valueSetTable customizeTableWrapper"
              >
                <table
                  class="usa-table usa-table--borderless customizeTable"
                  data-testid="table"
                >
                  <thead
                    class="display-flex flex-column valueSetTable__header"
                  >
                    <tr
                      class="valueSetTable__header_sectionHeader"
                    >
                      <th>
                        VALUE SET
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="overflowScroll valueSetTable__tableBody"
                  >
                    <tr
                      class="valueSetTable__tableBody_row"
                    >
                      <td>
                        <div
                          class="valueSetTable__tableBody_row_details"
                        >
                          <span
                            class="valueSetTable__tableBody_row_valueSetName"
                          >
                            <span
                              class=""
                            >
                              Cancer (Leukemia) Lab Result
                            </span>
                          </span>
                          <span
                            class="valueSetTable__tableBody_row_valueSetDetails"
                          >
                            <span
                              class=""
                            >
                              Cancer (Leukemia)  •   Labs  •  http://cap.org/eCC
                            </span>
                          </span>
                        </div>
                        <div>
                          <svg
                            aria-label="Right chevron indicating additional content"
                            class="usa-icon usa-icon--size-4"
                            focusable="false"
                            height="1em"
                            role="img"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                            />
                          </svg>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="valueSetTable__tableBody_row"
                    >
                      <td>
                        <div
                          class="valueSetTable__tableBody_row_details"
                        >
                          <span
                            class="valueSetTable__tableBody_row_valueSetName"
                          >
                            <span
                              class=""
                            >
                              Gonorrhea Medication
                            </span>
                          </span>
                          <span
                            class="valueSetTable__tableBody_row_valueSetDetails"
                          >
                            <span
                              class=""
                            >
                              Gonorrhea  •   Medications  •  RXNORM
                            </span>
                          </span>
                        </div>
                        <div>
                          <svg
                            aria-label="Right chevron indicating additional content"
                            class="usa-icon usa-icon--size-4"
                            focusable="false"
                            height="1em"
                            role="img"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                            />
                          </svg>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              class="content__right"
            />
          </div>
          <div
            class="paginationContainer"
          >
            <span>
              Showing 1 -
  2 of 
  2 value sets
            </span>
            <nav
              aria-label="Pagination"
              class="usa-pagination pagination"
            >
              <ul
                class="usa-pagination__list"
              >
                <li
                  class="usa-pagination__item usa-pagination__page-no"
                >
                  <button
                    aria-current="page"
                    aria-label="Page 1"
                    class="usa-button usa-button--unstyled usa-pagination__button usa-current"
                    data-testid="pagination-page-number"
                    type="button"
                  >
                    1
                  </button>
                </li>
              </ul>
            </nav>
            <div
              class="itemsPerPageContainer"
            >
              <label
                for="actionsPerPage"
              >
                Value sets per page
              </label>
              <select
                class="usa-select itemsPerPageDropdown"
                data-testid="Select"
                id="valeSetsPerPage"
                name="valeSetsPerPage"
              >
                <option
                  value="10"
                >
                  10
                </option>
                <option
                  value="25"
                >
                  25
                </option>
                <option
                  value="50"
                >
                  50
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
